src/gpu/optixPrograms.cu

```diff
@@ -148,9 +161,17 @@ extern "C" __global__ void __closesthit__()
 		return;
 	}
 
+	// Fix XYZ if distortion is applied (XYZ must be calculated in sensor coordinate frame)
+	if (ctx.doApplyDistortion) {
+		const int rayIdx = optixGetLaunchIndex().x;
+		Mat3x4f undistortedRay = ctx.rays[rayIdx];
+		Vec3f undistortedOrigin = undistortedRay * Vec3f{0, 0, 0};
```

Comment:
This line tells clearly that we need a method in mat library.
sth like `Transform.ToUnit()`